### PR TITLE
Add option for extra queries for SuperPmi

### PR DESCRIPTION
**Pull request by CarolEidt**
This adds an option to do extra queries for struct fields and types to make it easier to test and diff struct promotion enhancements, but we could consider adding other queries in future.

### CODE REVIEW

**Comment by CarolEidt** [comment]
@dotnet/jit-contrib PTAL

File: src/coreclr/src/jit/lclvars.cpp [hunk 1]
```diff
@@ -2670,7 +2670,49 @@ void Compiler::lvaSetStruct(unsigned varNum, CORINFO_CLASS_HANDLE typeHnd, bool
         compGSReorderStackLayout = true;
         varDsc->lvIsUnsafeBuffer = true;
     }
+#ifdef DEBUG
+    if (doExtraSuperPmiQueries)
```
Comment (BruceForstall) [inline src/coreclr/src/jit/lclvars.cpp:2674]: Do you need the `doExtraSuperPmiQueries` variable at all? Couldn't you just call `JitConfig.EnableExtraSuperPmiQueries()` here?

File: src/coreclr/src/jit/lclvars.cpp [hunk 2]
```diff
@@ -2670,7 +2670,49 @@ void Compiler::lvaSetStruct(unsigned varNum, CORINFO_CLASS_HANDLE typeHnd, bool
...
+    {
+        makeExtraStructQueries(typeHnd, 2);
```
Comment (BruceForstall) [inline src/coreclr/src/jit/lclvars.cpp:2676]: Presumably you could make `2` an option, but I guess it's fine to hard-code it.

File: src/coreclr/src/jit/lclvars.cpp [hunk 3]
```diff
@@ -2670,7 +2670,49 @@ void Compiler::lvaSetStruct(unsigned varNum, CORINFO_CLASS_HANDLE typeHnd, bool
...
+//------------------------------------------------------------------------
+// makeExtraStructQueries: Query the information for the given struct handle.
+//
+// Arguments:
+//    structHandle -- The handle for the struct type we're querying.
+//    level        -- How many more levels to recurse.
+//
+void Compiler::makeExtraStructQueries(CORINFO_CLASS_HANDLE structHandle, int level)
+{
+    if (level == 0)
```
Comment (BruceForstall) [inline src/coreclr/src/jit/lclvars.cpp:2691]: nit: since `level` is signed int, maybe this should be `level <= 0`

**Review by BruceForstall — APPROVED** [review]

File: src/coreclr/src/jit/lclvars.cpp (hunk 2 above)
Comment (CarolEidt) [reply src/coreclr/src/jit/lclvars.cpp:2676]: I thought about making it an option, but it didn't seem worth the complexity, albeit minor.

**Pull request merged**